app/router/assets/assets.go

```diff
@@ -0,0 +1,109 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package assets
+
+import (
+	"github.com/go-chi/chi"
+	"github.com/go-chi/render"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/service"
+	"github.com/limechain/hedera-eth-bridge-validator/app/model/asset"
+	"github.com/limechain/hedera-eth-bridge-validator/config/parser"
+	"github.com/limechain/hedera-eth-bridge-validator/constants"
+	"net/http"
+)
+
+var (
+	Route = "/assets"
+)
+
+type fungibleBridgeDetails struct {
+	*asset.FungibleAssetInfo
+	FeePercentage int64             `json:"feePercentage"`
+	MinAmount     string            `json:"minAmount"`
+	Networks      map[uint64]string `json:"networks"`
+}
+
+type nonFungibleBridgeDetails struct {
+	*asset.NonFungibleAssetInfo
+	Fee      int64             `json:"fee"`
+	Networks map[uint64]string `json:"networks"`
+}
+
+type networkAssets struct {
+	Fungible    map[string]fungibleBridgeDetails    `json:"fungible"`
+	NonFungible map[string]nonFungibleBridgeDetails `json:"nonFungible"`
+}
+
+// Router for assets
+func NewRouter(bridgeConfig parser.Bridge, assetsService service.Assets, pricingService service.Pricing) http.Handler {
+	r := chi.NewRouter()
+	r.Get("/", assetsResponse(bridgeConfig, assetsService, pricingService))
+	return r
+}
+
+// GET: .../assets
+func assetsResponse(bridgeConfig parser.Bridge, assetsService service.Assets, pricingService service.Pricing) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		responseContent := generateResponseContent(bridgeConfig, assetsService, pricingService)
+		render.JSON(w, r, responseContent)
+	}
+}
+
+func generateResponseContent(bridgeConfig parser.Bridge, assetsService service.Assets, pricingService service.Pricing) map[uint64]networkAssets {
+	response := make(map[uint64]networkAssets)
+
+	fungibleNetworkAssets := assetsService.FungibleNetworkAssets()
+	nonFungibleNetworkAssets := assetsService.NonFungibleNetworkAssets()
+	for networkId := range constants.NetworksById {
+		response[networkId] = networkAssets{
+			Fungible:    map[string]fungibleBridgeDetails{},
+			NonFungible: map[string]nonFungibleBridgeDetails{},
+		}
+
+		// Fungible
+		for _, assetAddress := range fungibleNetworkAssets[networkId] {
+			fungibleAssetInfo, existInfo := assetsService.FungibleAssetInfo(networkId, assetAddress)
+			minAmount, existMinAmount := pricingService.GetTokenPriceInfo(networkId, assetAddress)
+			bridgeTokenInfo := bridgeConfig.Networks[networkId].Tokens.Fungible[assetAddress]
+			if existInfo && existMinAmount {
+				fungibleAssetDetails := fungibleBridgeDetails{
+					FungibleAssetInfo: &fungibleAssetInfo,
+					FeePercentage:     bridgeTokenInfo.FeePercentage,
```

Comment:
If the token is wrapped, the fee percentage should be inherited from the native token